sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -521,6 +537,9 @@ CAFMaker::~CAFMaker()
   delete fFlatTree;
   delete fFlatFile;
 
+  //delete fGenieEvtRec;
+  //delete fGenieTree;
+
```

Comment:
Were these causing double-freeing or some other trouble?
Shouldn't the flat versions also be here?

Reply:
I pushed a commit where I now delete `fGenieEvtRec` and `fFlatGenieEvtRec`. But I found `fGenieTree` or `fFlatGenieTree` is deleted when the Record is delete.

Reply:
After investigation, the conclusion is that the trees are deleted when _the ROOT file objects_ they are associated to (e.g. `fFile`) are deleted.
In addition, the `fGenieEvtRec` _should_ be deleted by us since, according to [ROOT documentation ("If addr is not zero, but the pointer addr points at is zero" paragraph)](https://root.cern/doc/master/classTBranchElement.html#a7b19271c3bf833e6738bc18a43d6352f), we passed an address variable that has null content at branch address setting time, the record was allocated by ROOT but it still considered owned by us, and it should be explicitly deleted with `delete`.

Reply:
Thanks! In [1c47597](https://github.com/SBNSoftware/sbncode/pull/426/commits/1c47597b74548ba6589c79a543f99978130c9c30), I properly deleted all the GENIE objects.